inmem_store.go

```diff
@@ -131,3 +132,30 @@ func (i *InmemStore) GetUint64(key []byte) (uint64, error) {
 	defer i.l.RUnlock()
 	return i.kvInt[string(key)], nil
 }
+
+type commitIndexTrackingLog struct {
+	log         *Log
+	CommitIndex uint64
+}
+type InmemCommitTrackingStore struct {
+	InmemStore
+	commitIndex atomic.Uint64
+}
+
+// NewInmemCommitTrackingStore returns a new in-memory backend that tracks the commit index. Do not ever
+// use for production. Only for testing.
+func NewInmemCommitTrackingStore() *InmemCommitTrackingStore {
+	i := &InmemCommitTrackingStore{
+		InmemStore: *NewInmemStore(),
+	}
+	return i
+}
+
+func (i *InmemCommitTrackingStore) SetCommitIndex(index uint64) error {
```

Comment:
Oh great, I would also change the `NewInmemCommitTrackingStore` to return the `CommitTrackingStore` interface to prevent this kind of mistake😁

Reply:
returning interfaces is sometimes a bit of a pain because then you have to type assert if you need to use any method not in the interface. It might be fine.

The other common idiom for catching this kind of issues at compile time is to add this somewhere (usaully at top of file with constants):

```go

var _ CommitTrackingStore = &InmemCommitTrackingStore{}

```

That will be a compile error if the interface isn't implemented but is a no-op and doesn't actually allocate anything in real code because the result is unused.